api/src/main/java/org/apache/iceberg/actions/RewriteManifests.java

```diff
@@ -44,6 +47,38 @@ public interface RewriteManifests
    */
   RewriteManifests rewriteIf(Predicate<ManifestFile> predicate);
 
+  /**
+   * Rewrite manifests in a given order, based on partition field names
+   *
+   * <p>Supply an optional set of partition field names to sort the rewritten manifests by. Expects
+   * exact transformed column names used for partitioning; not the raw columnnames that partitions
+   * are derived from. E.G. supply 'data_bucket' and not 'data' for a bucket(N, data) partition
+   * definition
+   *
+   * <p>If not set, manifests will be rewritten in the order of the transforms in the table's
+   * current partition spec.
+   *
+   * @param partitionFieldSortOrder a list of partition field names
+   * @return this for method chaining
+   */
+  default RewriteManifests sort(List<String> partitionFieldSortOrder) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " doesn't implement sort(List<String>)");
+  }
+
+  /**
+   * Rewrite manifests in a given order, dictated by a custom Function
+   *
+   * <p>Supply a Function which will apply its own custom clustering logic based on supplied {@link
+   * org.apache.iceberg.DataFile} attributes.
+   *
+   * @param sortStrategyFunction A Function that returns a String to be used for manifest clustering
+   * @return this method for chaining
+   */
+  default RewriteManifests sort(Function<DataFile, String> sortStrategyFunction) {
+    throw new UnsupportedOperationException();
```

Comment:
this needs a msg

Reply:
Ack, my bad

Reply:
I think this is not resolved yet?

Reply:
I haven't pushed this commit yet. Will shortly